lib/json/JsonGenerator.js

```diff
@@ -164,7 +164,7 @@ class JsonGenerator extends Generator {
 		const jsonStr = stringifySafe(finalJson);
 		const jsonExpr =
 			jsonStr.length > 20 && typeof finalJson === "object"
-				? `JSON.parse(${JSON.stringify(jsonStr)})`
+				? `JSON.parse('${jsonStr.replace(/[\\']/g, "\\$&")}')`
```

Comment:
newlines should also be escaped as they would break the single quoted string into multiple lines making it invalid.

Reply:
though stringifySafe does not permit newlines, so that would not happen - but then test is needed to ensure it is not regressed in the future.

Reply:
since this is the output of `JSON.stringify` it won't contain newlines and most other weird stuff is already escaped.

Reply:
We have tests with new lines in JSON, so I guess this should already be covered